browser/ui/views/text_recognition_dialog_view.cc

```diff
@@ -0,0 +1,126 @@
+#include "ui/views/controls/scroll_view.h"
+#include "ui/views/layout/flex_layout.h"
+#include "ui/views/widget/widget.h"
+#include "ui/views/window/dialog_client_view.h"
+
+namespace brave {
+
+void ShowTextRecognitionDialog(content::WebContents* web_contents,
+                               const SkBitmap& image) {
+  constrained_window::ShowWebModalDialogViews(
+      new TextRecognitionDialogView(image), web_contents)
+      ->Show();
+}
+
+}  // namespace brave
+
+TextRecognitionDialogView::TextRecognitionDialogView(const SkBitmap& image) {
+  SetModalType(ui::MODAL_TYPE_CHILD);
+  SetButtons(ui::DIALOG_BUTTON_OK);
+  SetButtonLabel(ui::DIALOG_BUTTON_OK,
+                 brave_l10n::GetLocalizedResourceUTF16String(
+                     IDS_TEXT_RECOG_DIALOG_CLOSE_BUTTON));
+  SetShowCloseButton(false);
+
+  SetLayoutManager(std::make_unique<views::FlexLayout>())
+      ->SetOrientation(views::LayoutOrientation::kVertical)
+      .SetMainAxisAlignment(views::LayoutAlignment::kStart)
+      .SetInteriorMargin(gfx::Insets::TLBR(24, 26, 0, 26));
+
+  header_label_ = AddChildView(std::make_unique<views::Label>());
+  const int size_diff = 14 - views::Label::GetDefaultFontList().GetFontSize();
+  header_label_->SetFontList(
+      views::Label::GetDefaultFontList()
+          .DeriveWithSizeDelta(size_diff)
+          .DeriveWithWeight(gfx::Font::Weight::SEMIBOLD));
+  header_label_->SetHorizontalAlignment(gfx::ALIGN_LEFT);
+  header_label_->SetProperty(views::kMarginsKey,
+                             gfx::Insets::TLBR(0, 0, 10, 0));
+
+  if (image.empty()) {
+    UpdateContents({});
+    return;
+  }
+
+  StartExtractingText(image);
+}
+
+TextRecognitionDialogView::~TextRecognitionDialogView() = default;
+
+void TextRecognitionDialogView::StartExtractingText(const SkBitmap& image) {
+  header_label_->SetText(brave_l10n::GetLocalizedResourceUTF16String(
+      IDS_TEXT_RECOG_DIALOG_HEADER_IN_PROGRESS));
+
+  base::ThreadPool::PostTaskAndReplyWithResult(
+      FROM_HERE,
+      {base::MayBlock(), base::TaskShutdownBehavior::SKIP_ON_SHUTDOWN},
+      base::BindOnce(&GetTextFromImage, image),
+      base::BindOnce(&TextRecognitionDialogView::OnGetTextFromImage,
+                     weak_factory_.GetWeakPtr()));
+}
+
+void TextRecognitionDialogView::OnGetTextFromImage(
+    const std::vector<std::string>& text) {
+  UpdateContents(text);
+  AdjustWidgetSize();
+}
+
+void TextRecognitionDialogView::UpdateContents(
+    const std::vector<std::string>& text) {
+  if (text.empty()) {
+    header_label_->SetText(brave_l10n::GetLocalizedResourceUTF16String(
+        IDS_TEXT_RECOG_DIALOG_HEADER_FAILED));
+    return;
+  }
+
+  header_label_->SetText(brave_l10n::GetLocalizedResourceUTF16String(
+      IDS_TEXT_RECOG_DIALOG_HEADER_COMPLETE));
+
+  // Treat each string in |text| as a separated line string.
+  std::string unified_string;
+  for (const auto& t : text) {
+    unified_string += t;
+    unified_string += '\n';
+  }
```

Comment:
The above is fine 😄 But if you could use std library:
```C++
const char* const delim = "\n";
std::ostringstream unified;
std::copy(text.begin(), text.end(),
           std::ostream_iterator<std::string>(unified, delim));
std::string unified_string = unified.str();
```

Need to include <sstream> and maybe some others (lint should warn with include what you use)

Reply:
Pushed a quick update with this - https://github.com/brave/brave-core/pull/16508/commits/1d611fed32399e2c4f730008db6d3a4d8df9d1d2 😄

Reply:
Inspired from your changes - I found `base::JoinString()` api!

Reply:
Sometimes I don't understand why we don't have basic operation like join string in std 😭

Reply:
Nice find! 😃